Replace usage of MUI date picker with Oxygen UI date picker

amandasiriwardane opened the pull request:
<!-- Please fill in the pull request template when submitting pull requests. -->

### Purpose
This PR updates the DatePicker component in the form module to use the Oxygen UI wrapper instead of importing directly from MUI, ensuring consistency with the design system.



### Related Issues
- Resolves wso2/product-is#25669


### Related PRs
- Related to #9118


### Checklist

- [ ] e2e cypress tests locally verified. (for internal contributers)
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [x] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [x] Ran FindSecurityBugs plugin and verified report
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

## Developer Checklist (Mandatory)
- [x] Complete the **Developer Checklist** in the related `product-is` issue to track any behavioral change or migration impact.

### Additional Context

**Technical Details:**
The Oxygen UI DatePicker (v2.4.6) is a wrapper around MUI's DatePicker component that provides consistent theming and styling across the Identity Server applications. The original implementation imported directly from MUI, bypassing the design system.

**Type Handling:**
Type assertions were necessary because the Oxygen UI DatePicker uses a generic type parameter `TEnableAccessibleFieldDOMStructure` which defaults to `true`. To maintain compatibility with the existing `moment.Moment` date handling, type assertions were applied to the rest props.

**Build Verification:**
- All 13 projects in the monorepo build successfully (13m build time)
- No TypeScript compilation errors
- No runtime errors expected as the Oxygen UI DatePicker is a direct wrapper of the MUI component 
-Webpack dev server runs without errors

**Impact:**
- No behavioral changes - the DatePicker functionality remains identical
- No migration impact - this is an internal implementation change
- No configuration changes required

**Note**: As an external contributor without the full Identity Server backend setup, manual UI testing was not performed. However, since the Oxygen UI DatePicker is a direct wrapper of the MUI DatePicker with identical props and behavior, functionality remains unchanged. Build validation confirms the code is correct.
<img width="1268" height="259" alt="Screenshot 2025-10-19 141910" src="https://github.com/user-attachments/assets/c3aeee90-8f9b-4754-a384-4665882b3de1" />
<img width="1224" height="492" alt="Screenshot 2025-10-19 145214" src="https://github.com/user-attachments/assets/612f65c0-f9fc-49b4-a699-51d5f025ac63" />

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/wso2/identity-apps?pullRequest=9367) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/wso2/identity-apps?pullRequest=9367) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/wso2/identity-apps?pullRequest=9367) it.</sub>

brionmario commented on apps/myaccount/java/org.wso2.identity.apps.myaccount.server.feature/src/main/resources/build.properties:
Is this interntional?

amandasiriwardane committed:
fix: Use Oxygen UI DatePicker wrapper instead of direct MUI import
Fixes #25669

- Updated DatePicker import to use @oxygen-ui/react/DatePicker
- Added type assertions to handle Oxygen UI DatePicker generic types
- Improved type safety with moment.isMoment() check in onChange handler
